### PR TITLE
Distribution: Add PiecewiseLinearDistribution

### PR DESCRIPTION
**Pull request by regislebrun**
Add a new 1D continuous distribution in the experimental namespace whose probability density function is a piecewise linear function defined by abscissae x and ordinates y.

Key characteristics:
- The PDF is linearly interpolated between the given knot points and zero outside the support [x_0, x_{n-1}].
- The ordinates y are nonnegative and automatically rescaled so that the total integral of the PDF equals 1.
- Segment integrals are precomputed for efficient CDF and probability computations.
- Quantile computation uses a quadratic formula for segment inversion with support for both direct and tail probabilities.
- Moments (mean, variance, skewness, kurtosis), entropy, roughness, and PDF singularities are computed analytically using Kahan compensated summation for numerical accuracy.
- Serialization via save/load (XML Study) and Python pickle is supported.
- Full SWIG bindings with docstrings in the experimental module.

## Summary by CodeRabbit

- **New Features**
  - Added the experimental `PiecewiseLinearDistribution` for continuous distributions defined by support points and nonnegative density values.
  - Supports sampling, PDF/CDF evaluation, quantiles, moments, probabilities, entropy, roughness, and parameter access.
  - Truncation preserves the piecewise-linear form when applicable.
  - Available through Python, including plotting examples.

- **Documentation**
  - Added user guidance, API documentation, parameter descriptions, and usage examples.

- **Tests**
  - Added comprehensive validation for calculations, sampling, serialization, plotting, numerical consistency, and error handling.

### CODE REVIEW

File: python/src/PiecewiseLinearDistribution_doc.i [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+Parameters
+----------
+x : sequence of float, size :math:`n \geq 2`
+    Abscissae, strictly increasing.
+
+    Default value is [-1.0, 1.0].
+y : sequence of float, size :math:`n`
+    Ordinates (PDF values at the abscissae), nonnegative.
+
+    Default value is [0.5, 0.5].
```
Comment (jschueller) [inline python/src/PiecewiseLinearDistribution_doc.i]: misleading: x,y arguments have no default values, its the default ctor that sets this pair of x,y values

File: python/src/PiecewiseLinearDistribution_doc.i [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+"Entropy of the distribution.
+
+Returns
+-------
+entropy : float
+    The entropy of the distribution."
+
+// ---------------------------------------------------------------------
+
+%feature("docstring") OT::PiecewiseLinearDistribution::getRoughness
```
Comment (jschueller) [inline python/src/PiecewiseLinearDistribution_doc.i:119]: no need to override these functions just to document the formulas: rather include them in the body of the main docstring (or eventually in the Notes section)

File: lib/src/Uncertainty/Distribution/PiecewiseLinearDistribution.cxx [hunk 3]
```diff
@@ -0,0 +1,614 @@
...
+
+/* Parameters value accessors */
+Point PiecewiseLinearDistribution::getParameter() const
+{
+  const UnsignedInteger n = x_.getSize();
+  Point parameter(2 * n);
+  for (UnsignedInteger i = 0; i < n; ++i)
+  {
+    parameter[i] = x_[i];
+    parameter[n + i] = y_[i];
```
Comment (jschueller) [inline lib/src/Uncertainty/Distribution/PiecewiseLinearDistribution.cxx]: use std::copy

File: lib/src/Uncertainty/Distribution/PiecewiseLinearDistribution.cxx [hunk 4]
```diff
@@ -0,0 +1,614 @@
...
+{
+  const UnsignedInteger n = x_.getSize();
+  if (parameter.getSize() != 2 * n) throw InvalidArgumentException(HERE) << "Error: expected " << 2 * n << " values, got " << parameter.getSize();
+  const Scalar w = getWeight();
+  Point x(n);
+  Point y(n);
+  for (UnsignedInteger i = 0; i < n; ++i)
+  {
+    x[i] = parameter[i];
+    y[i] = parameter[n + i];
```
Comment (jschueller) [inline lib/src/Uncertainty/Distribution/PiecewiseLinearDistribution.cxx]: use std::copy

File: lib/src/Uncertainty/Distribution/PiecewiseLinearDistribution.cxx [hunk 5]
```diff
@@ -0,0 +1,614 @@
...
+
+/* Parameters description accessor */
+Description PiecewiseLinearDistribution::getParameterDescription() const
+{
+  const UnsignedInteger n = x_.getSize();
+  Description description(2 * n);
+  for (UnsignedInteger i = 0; i < n; ++i)
+  {
+    description[i] = OSS() << "x" << i;
+    description[n + i] = OSS() << "y" << i;
```
Comment (jschueller) [inline lib/src/Uncertainty/Distribution/PiecewiseLinearDistribution.cxx]: use Description::BuildDefault

**Comment by jschueller** [comment]
issue 3135 is unrelated

File: lib/src/Uncertainty/Distribution/openturns/PiecewiseLinearDistribution.hxx [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+
+#include "openturns/OTprivate.hxx"
+#include "openturns/DistributionImplementation.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+/**
+ * @class PiecewiseLinearDistribution
+ *
+ * The PiecewiseLinearDistribution distribution.
```
Comment (jschueller) [inline lib/src/Uncertainty/Distribution/openturns/PiecewiseLinearDistribution.hxx:32]: todo: add a special case in truncateddistribution

**Comment by regislebrun** [comment]
fixed

Addressed all review comments:
- Fixed unsigned underflow on empty x (probabilities_ init to 0)
- Fixed quantile inversion: divide residual by segment width dx
- Fixed mean_ access: use getMean() in computeCovariance, getSkewness, getKurtosis
- Fixed setY: reject all-zero ordinate vectors
- Used std::copy in getParameter/setParameter
- Used Description::BuildDefault in getParameterDescription
- Fixed load() with x size guard
- Fixed doc: removed misleading default values for x,y params
- Removed redundant computeEntropy/getRoughness doc overrides
- Added non-unit-width quantile round-trip tests
- Fixed pyplot: shift sin by 1.5 for strictly positive ordinates

Note: computeRange() left as-is (infinite bounds) since DistributionValidation assumes PDF=0 at finite bounds, but PiecewiseLinearDistribution can have non-zero PDF at the boundary.

File: lib/src/Uncertainty/Distribution/PiecewiseLinearDistribution.cxx (hunk 3 above)
Comment (regislebrun) [reply lib/src/Uncertainty/Distribution/PiecewiseLinearDistribution.cxx]: fixed

File: lib/src/Uncertainty/Distribution/PiecewiseLinearDistribution.cxx (hunk 4 above)
Comment (regislebrun) [reply lib/src/Uncertainty/Distribution/PiecewiseLinearDistribution.cxx]: fixed

File: lib/src/Uncertainty/Distribution/PiecewiseLinearDistribution.cxx (hunk 5 above)
Comment (regislebrun) [reply lib/src/Uncertainty/Distribution/PiecewiseLinearDistribution.cxx]: fixed

File: lib/src/Uncertainty/Distribution/openturns/PiecewiseLinearDistribution.hxx (hunk 6 above)
Comment (regislebrun) [reply lib/src/Uncertainty/Distribution/openturns/PiecewiseLinearDistribution.hxx:32]: fixed

File: lib/src/Uncertainty/Distribution/PiecewiseLinearDistribution.cxx [hunk 7]
```diff
@@ -0,0 +1,614 @@
...
+}
+
+/* Parameters constructor */
+PiecewiseLinearDistribution::PiecewiseLinearDistribution(const Point & x,
+    const Point & y)
+  : DistributionImplementation()
+  , x_(x)
+  , y_(y)
+  , yNorm_(y)
+  , probabilities_(x.getSize() - 1)
```
[the remark this answers is not in the document]
Comment (regislebrun) [reply lib/src/Uncertainty/Distribution/PiecewiseLinearDistribution.cxx]: fixed

File: lib/src/Uncertainty/Distribution/PiecewiseLinearDistribution.cxx [hunk 8]
```diff
@@ -0,0 +1,614 @@
...
+  isAlreadyComputedCovariance_ = false;
+  computeRange();
+}
+
+/* Compute the numerical range of the distribution given the parameters values */
+void PiecewiseLinearDistribution::computeRange()
+{
+  Interval range(x_[0], x_[x_.getSize() - 1]);
+  range.setFiniteLowerBound(Interval::BoolCollection(1, false));
+  range.setFiniteUpperBound(Interval::BoolCollection(1, false));
```
[the remark this answers is not in the document]
Comment (regislebrun) [reply lib/src/Uncertainty/Distribution/PiecewiseLinearDistribution.cxx:176]: fixed

File: lib/src/Uncertainty/Distribution/PiecewiseLinearDistribution.cxx [hunk 9]
```diff
@@ -0,0 +1,614 @@
...
+        // Linear case: b*t = remain
+        t = (b > 0.0) ? remain / b : 0.0;
+      }
+      else
+      {
+        // Quadratic: a*t^2 + b*t - remain = 0
+        const Scalar discriminant = b * b + 4.0 * a * remain;
+        t = (-b + std::sqrt(std::max(0.0, discriminant))) / (2.0 * a);
+        if (t < 0.0) t = 0.0;
+        if (t > 1.0) t = 1.0;
```
[the remark this answers is not in the document]
Comment (regislebrun) [reply lib/src/Uncertainty/Distribution/PiecewiseLinearDistribution.cxx]: fixed

File: lib/src/Uncertainty/Distribution/PiecewiseLinearDistribution.cxx [hunk 10]
```diff
@@ -0,0 +1,614 @@
...
+    const Scalar x1 = x_[i + 1];
+    const Scalar x0_2 = x0 * x0;
+    const Scalar x1_2 = x1 * x1;
+    const Scalar y0 = yNorm_[i];
+    const Scalar y1 = yNorm_[i + 1];
+    m2Sum.add(dx * (y0 * (3.0 * x0_2 + 2.0 * x0 * x1 + x1_2)
+                     + y1 * (x0_2 + 2.0 * x0 * x1 + 3.0 * x1_2)) / 12.0);
+  }
+  covariance_ = CovarianceMatrix(1);
+  covariance_(0, 0) = m2Sum.value - mean_[0] * mean_[0];
```
[the remark this answers is not in the document]
Comment (regislebrun) [reply lib/src/Uncertainty/Distribution/PiecewiseLinearDistribution.cxx]: fixed

File: lib/src/Uncertainty/Distribution/PiecewiseLinearDistribution.cxx [hunk 11]
```diff
@@ -0,0 +1,614 @@
...
+
+/* Y accessor */
+void PiecewiseLinearDistribution::setY(const Point & y)
+{
+  if (y.getSize() != y_.getSize()) throw InvalidArgumentException(HERE) << "Error: new y must have the same size as the current y";
+  for (UnsignedInteger i = 0; i < y.getSize(); ++i)
+  {
+    if (y[i] < 0.0)
+      throw InvalidArgumentException(HERE) << "Error: y must be nonnegative";
+  }
```
[the remark this answers is not in the document]
Comment (regislebrun) [reply lib/src/Uncertainty/Distribution/PiecewiseLinearDistribution.cxx:562]: fixed

File: python/src/PiecewiseLinearDistribution_doc.i (hunk 1 above)
Comment (regislebrun) [reply python/src/PiecewiseLinearDistribution_doc.i]: fixed

File: python/src/PiecewiseLinearDistribution_doc.i (hunk 2 above)
Comment (regislebrun) [reply python/src/PiecewiseLinearDistribution_doc.i:119]: fixed

File: lib/src/Uncertainty/Distribution/openturns/PiecewiseLinearDistribution.hxx (hunk 6 above)
Comment (regislebrun) [reply lib/src/Uncertainty/Distribution/openturns/PiecewiseLinearDistribution.hxx:32]: fixed: added PiecewiseLinearDistribution special case in TruncatedDistribution::buildSimplifiedVersion. Clips x/y arrays to the truncation bounds with linear interpolation at new boundaries.

File: lib/src/Uncertainty/Distribution/PiecewiseLinearDistribution.cxx [hunk 12]
```diff
@@ -0,0 +1,612 @@
...
+/* Method load() reloads the object from the StorageManager */
+void PiecewiseLinearDistribution::load(Advocate & adv)
+{
+  DistributionImplementation::load(adv);
+  adv.loadAttribute("x_", x_);
+  adv.loadAttribute("y_", y_);
+  if (x_.getSize() < 2)
+    throw InvalidArgumentException(HERE) << "Error: x must have at least 2 elements, here x.getSize()=" << x_.getSize();
+  yNorm_ = y_;
+  probabilities_ = Point(x_.getSize() - 1);
```
[the remark this answers is not in the document]
Comment (regislebrun) [reply lib/src/Uncertainty/Distribution/PiecewiseLinearDistribution.cxx:623]: fixed: load() now validates y_.getSize() == x_.getSize() before calling update(), matching the constructor invariant.

File: lib/src/Uncertainty/Distribution/TruncatedDistribution.cxx [hunk 13]
```diff
@@ -356,6 +357,68 @@ Bool TruncatedDistribution::hasSimplifiedVersion(Distribution & simplified) cons
...
+      const PiecewiseLinearDistribution * pwd(dynamic_cast< const PiecewiseLinearDistribution * >(localDistribution.getImplementation().get()));
+      const Point xOld(pwd->getX());
+      const Point yOld(pwd->getY());
+      const Scalar a = xOld[0];
+      const Scalar b = xOld[xOld.getSize() - 1];
+      // Collect x and y values within [alpha, beta], interpolating at boundaries
+      Point xNew;
+      Point yNew;
+      // Interpolate at alpha if it falls inside the original support
+      if (alpha > a)
```
[the remark this answers is not in the document]
Comment (regislebrun) [reply lib/src/Uncertainty/Distribution/TruncatedDistribution.cxx]: fixed - Changed boundary conditions from strict to inclusive (`>` to `>=`, `<` to `<=`) so endpoints equal to original support bounds are retained. Added regression test comparing simplified vs truncated PDF/CDF at boundary values.

**Comment by regislebrun** [comment]
fixed

Addressed all remaining review issues:

1. **load() validation**: Added validation for non-increasing x values, negative y values, and all-zero y vectors in load() before calling update(). This prevents malformed persisted data from producing invalid distributions.

2. **Closes #3135 reference removed**: Removed incorrect "Closes openturns#3135" from commit message. Issue #3135 is about SIGINT handler restoration and is unrelated to PiecewiseLinearDistribution.

3. **Docstring coverage**: Added docstrings for computeEntropy, getRoughness, getParameter, setParameter, getParameterDescription, computeScalarQuantile, and computeProbability to reach the 80% coverage threshold.

File: python/src/PiecewiseLinearDistribution_doc.i [hunk 14]
```diff
@@ -0,0 +1,195 @@
...
+
+Parameters
+----------
+interval : :class:`~openturns.Interval`
+    The interval of dimension 1.
+
+Returns
+-------
+probability : float
+    The probability :math:`P(X \\in \\text{interval})`."
```
Comment (jschueller) [inline python/src/PiecewiseLinearDistribution_doc.i:195]: please drop the redundant docstring overrides

File: python/test/t_PiecewiseLinearDistribution_std.py [hunk 15]
```diff
@@ -494,7 +494,111 @@
...
+dist_test = otexp.PiecewiseLinearDistribution(x, y)
+with ott.assert_raises(TypeError):
+    dist_test.setY(ot.Point([0.0, 0.0, 0.0]))
+print("OK: setY all-zero after construction thrown")
+
+# ---------------------------------------------------------------------------
+# 29. load() validation -- corrupted XML data
+# ---------------------------------------------------------------------------
+# Create valid distribution, save to XML, tamper, reload
+tmpfile = tempfile.mktemp(suffix=".xml")
```
Comment (jschueller) [inline python/test/t_PiecewiseLinearDistribution_std.py:534]: this test is a bit overkill, and not really robust

File: python/src/PiecewiseLinearDistribution_doc.i [hunk 16]
```diff
@@ -0,0 +1,195 @@
...
+"Accessor to the distribution's ordinates.
+
+Parameters
+----------
+y : sequence of float, size :math:`n`
+    Ordinates (PDF values at the abscissae), nonnegative."
+
+// ---------------------------------------------------------------------
+
+%feature("docstring") OT::PiecewiseLinearDistribution::getSingularities
```
Comment (jschueller) [inline python/src/PiecewiseLinearDistribution_doc.i:96]: drop all the docstring overrides below

**Review by jschueller — APPROVED** [review]

File: python/test/t_PiecewiseLinearDistribution_std.py [hunk 17]
```diff
@@ -0,0 +1,608 @@
...
+    pdf_val = distribution.computePDF(test_x)
+    logpdf_val = distribution.computeLogPDF(test_x)
+    if pdf_val > 0.0:
+        ott.assert_almost_equal(logpdf_val, math.log(pdf_val))
+    else:
+        ott.assert_almost_equal(logpdf_val, float("-inf"))
+print("LogPDF OK")
+
+# ---------------------------------------------------------------------------
+# 25. TruncatedDistribution endpoint regression
```
Comment (jschueller) [inline python/test/t_PiecewiseLinearDistribution_std.py:468]: the other simplification tests are all in the TruncatedDistribution test, maybe we can move these there too for consistency

File: python/test/t_PiecewiseLinearDistribution_std.py (hunk 15 above)
Comment (regislebrun) [reply python/test/t_PiecewiseLinearDistribution_std.py]: fixed

File: python/src/PiecewiseLinearDistribution_doc.i (hunk 16 above)
Comment (regislebrun) [reply python/src/PiecewiseLinearDistribution_doc.i:96]: fixed

File: python/test/t_PiecewiseLinearDistribution_std.py (hunk 17 above)
Comment (regislebrun) [reply python/test/t_PiecewiseLinearDistribution_std.py]: fixed

**Review by adutfoy — APPROVED** [review]
Good for me! Nice distribution!!!

**Comment by regislebrun** [comment]
@jschueller two green, ready to go into master!
